appinventor/blocklyeditor/src/blocks/math.js

```diff
@@ -987,16 +987,131 @@ Blockly.Blocks['math_format_as_decimal'] = {
 };
 
 Blockly.Blocks['math_is_a_number'] = {
-  category: 'Math',
-  helpUrl: Blockly.Msg.LANG_MATH_IS_A_NUMBER_HELPURL,
-  init: function () {
+  category : Blockly.Msg.LANG_CATEGORY_MATH,
+  helpUrl: function() {
+      var mode = this.getFieldValue('OP');
+      return Blockly.Blocks.math_is_a_number.HELPURLS[mode];
+  },
+  init : function() {
     this.setColour(Blockly.MATH_CATEGORY_HUE);
-    this.setOutput(true, Blockly.Blocks.Utilities.YailTypeToBlocklyType("boolean", Blockly.Blocks.Utilities.OUTPUT));
+    this.setOutput(true, Blockly.Blocks.Utilities.YailTypeToBlocklyType("boolean",Blockly.Blocks.Utilities.OUTPUT));
     this.appendValueInput('NUM')
-        .appendField(Blockly.Msg.LANG_MATH_IS_A_NUMBER_INPUT_NUM);
-    this.setTooltip(function () {
-      return Blockly.Msg.LANG_MATH_IS_A_NUMBER_TOOLTIP;
+        .appendField(new Blockly.FieldDropdown(this.OPERATORS), 'OP');
+    // Assign 'this' to a variable for use in the closures below.
+    var thisBlock = this;
+    this.setTooltip(function() {
+      var mode = thisBlock.getFieldValue('OP');
+      return Blockly.Blocks.math_is_a_number.TOOLTIPS[mode];
+    });
+  },
+  typeblock: [{
+    translatedName: Blockly.Msg.LANG_MATH_IS_A_NUMBER_INPUT_NUM,
+    dropDown: {
+      titleName: 'OP',
+      value: 'NUMBER'
+    }
+  },{
+    translatedName: Blockly.Msg.LANG_MATH_IS_A_DECIMAL_INPUT_NUM,
+    dropDown: {
+      titleName: 'OP',
+      value: 'DECIMAL'
+    }
+  },{
+    translatedName: Blockly.Msg.LANG_MATH_IS_A_HEXADECIMAL_INPUT_NUM,
+    dropDown: {
+      titleName: 'OP',
+      value: 'HEXADECIMAL'
+    }
+  },{
+    translatedName: Blockly.Msg.LANG_MATH_IS_A_BINARY_INPUT_NUM,
+    dropDown: {
+      titleName: 'OP',
+      value: 'BINARY'
+    }
+  }]
+};
+
+Blockly.Blocks.math_is_a_number.OPERATORS =
+  [[ Blockly.Msg.LANG_MATH_IS_A_NUMBER_INPUT_NUM, 'NUMBER' ],
+   [ Blockly.Msg.LANG_MATH_IS_A_DECIMAL_INPUT_NUM, 'DECIMAL' ],
```

Comment:
This is a confusing name for this test.   If x is 43.2 than is-number?  is true, and is-decimal? is false.  Most people would say that 43.2 is a decimal.   I realize you mean "decimal integer" and maybe that's what to use here.   Or may just is-base-10 ?